Shared/Samples/Edit geodatabase with transactions/EditGeodatabaseWithTransactionsView.swift

```diff
@@ -0,0 +1,266 @@
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct EditGeodatabaseWithTransactionsView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The action currently being preformed.
+    @State private var selectedAction: Action? = .setUp
+    
+    /// The text describing the status of the sample.
+    @State private var statusText = ""
+    
+    /// The point on the map where the user tapped.
+    @State private var tapLocation: Point?
+    
+    /// A Boolean value indicating whether a transaction is active on the geodatabase.
+    @State private var isInTransaction = false
+    
+    /// A Boolean value indicating whether a transaction is required to add a feature.
+    @State private var transactionIsRequired = true
+    
+    /// A Boolean value indicating whether the select feature type popover is presented.
+    @State private var isSelectingFeatureType = false
+    
+    /// A Boolean value indicating whether the alert to end a transaction is presented.
+    @State private var endTransactionAlertIsPresented = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapView(map: model.map)
+            .onSingleTapGesture { _, mapPoint in
+                // Shows the select feature type popover when a feature can currently be added.
+                guard !transactionIsRequired || isInTransaction else { return }
+                
+                tapLocation = mapPoint
+                isSelectingFeatureType = true
+            }
+            .task(id: selectedAction) {
+                guard let selectedAction else { return }
+                
+                do {
+                    switch selectedAction {
+                    case .setUp:
+                        try await model.setUp()
+                    case .addFeature(let tableName, let featureTypeID):
+                        try await model.addFeature(
+                            to: tableName,
+                            featureTypeID: featureTypeID,
+                            point: tapLocation!
+                        )
+                    case .beginTransaction:
+                        try model.geodatabase.beginTransaction()
+                        isInTransaction = true
+                    case .commitTransaction:
+                        try model.geodatabase.commitTransaction()
+                        isInTransaction = false
+                    case .rollbackTransaction:
+                        try model.geodatabase.rollbackTransaction()
+                        isInTransaction = false
+                    case .sync:
+                        try await model.syncGeodatabase()
+                    }
+                    
+                    statusText = selectedAction.completionMessage
+                } catch {
+                    self.error = error
```

Comment:
I'm wondering if the type of the error can be narrowed down to `GeodatabaseError`, or capture the `catch let someError as GeodatabaseError` first, before the generic error, to show some details in the status text. Thus, it makes it easier for the user to see what is the error really about.

Reply:
That makes sense to me. I'll consider revisiting it during Sprint X, but I went with your original suggestion for now.